src/NuGet.Core/NuGet.Versioning/VersionComparer.cs

```diff
@@ -348,7 +348,7 @@ private static string[] GetReleaseLabelsOrNull(SemanticVersion version)
                 var enumerable = version.ReleaseLabels;
                 labels = enumerable as string[];
```

Comment:
Why we're expecting `string[]` here? When does it happen and when it doesn't?

Reply:
Is it possible to replace `IEnumerable[]` type of `ReleaseLabels`  with `string[]`? 
For example we're taking Count for `IEnumerable` it have to iterate all items but for `string[]` it's already calculated there. Of course it needs much more code refactoring maybe not now if it's takes much more time.


https://github.com/NuGet/NuGet.Client/blob/fcc4bc40aa1c123d003dea822bd5c26fd68cefe4/src/NuGet.Core/NuGet.Versioning/VersionRange.cs#L347

Reply:
no, it's part of the public API and therefore the ABI as well. Changing it will break everyone who uses a different package compiled against an older version of `NuGet.Versioning.dll` (and calls the `ReleaseLabels` getter), but then the project restore graph bring in the new version of `NuGet.Versioning`. There will be `MethodNotFoundException` at runtime.

If we could change it, we could get more performance improvements, but it's a breaking ABI change.